modules/nlu/src/backend/module-lifecycle/on-server-started.ts

```diff
@@ -54,25 +59,22 @@ const registerMiddleware = async (bp: typeof sdk, state: NLUState) => {
         return next(undefined, false, true)
       }
 
-      let nluResults = {}
+      let nluResults: sdk.IO.EventUnderstanding
       const { engine } = state.nluByBot[event.botId]
       const extractEngine2 = async () => {
-        try {
-          // eventually if model not loaded for bot languages ==> train or load
+        // eventually if model not loaded for bot languages ==> train or load
+        nluResults = await engine.predict(event.preview, event.nlu.includedContexts)
+        if (nluResults.errored && nluResults.suggestedLanguage) {
+          const model = await getLatestModel(bp.ghost.forBot(event.botId), nluResults.suggestedLanguage)
```

Comment:
is suggestedLanguage always part of supported languages for given botId ? is it possible that we land on a situation where no model has been trained for suggested language ? in that case, model will be null and make engine.load crash.

Reply:
You're right, but this was always the case. Its the previous behaviour.

I suggest we add a TODO comment for now and handle this case better in the future. This part of the code is about to change anyway for something like this:

```
let language = state.nluByBot[event.botId].defaultLanguage
nluResults = await engine.predict(event.preview, event.nlu.includedContexts, language)
if (nluResults.suggestedLanguage !== language) {
    language = nluResults.suggestedLanguage
    nluResults = await engine.predict(event.preview, event.nlu.includedContexts, language)
} else if (nluResults.error === sdk.NLU.ErrorStatus.NO_PREDICTOR_LOADED) {
    const model = await getLatestModel(ghost, language)
   // assert model is not null. If it is, handle error correctly
   nluResults = await engine.predict(event.preview, event.nlu.includedContexts, language)
} else if (nluResults.error) {
    throw and repport error correctly
}
```